src/main/webapp/app/exercises/quiz/manage/quiz-exercise-detail.component.html

```diff
@@ -207,83 +207,14 @@ <h4>
-                        <span jhiTranslate="artemisApp.quizExercise.addMultipleChoiceQuestion"></span>
-                    </button>
-                </div>
-                <div class="col-12 col-sm-6 col-xl-3 mb-1">
-                    <button id="quiz-add-dnd-question" class="btn btn-block btn-success" (click)="addDragAndDropQuestion()">
-                        <fa-icon [icon]="faPlus"></fa-icon>
-                        <span jhiTranslate="artemisApp.quizExercise.addDragAndDropQuestion"></span>
-                    </button>
-                </div>
-                <div class="col-12 col-sm-6 col-xl-3 mb-1">
-                    <button id="quiz-add-short-answer-question" class="btn btn-block btn-success" (click)="addShortAnswerQuestion()">
-                        <fa-icon [icon]="faPlus"></fa-icon>
-                        <span jhiTranslate="artemisApp.quizExercise.addShortAnswerQuestion"></span>
-                    </button>
-                </div>
-                <div class="col-12 col-sm-6 col-xl-3 mb-1">
-                    <button class="btn btn-block" (click)="showHideExistingQuestions()" [ngClass]="{ 'btn-primary': showExistingQuestions, 'btn-success': !showExistingQuestions }">
-                        <fa-icon [icon]="faPlus"></fa-icon>
-                        <span jhiTranslate="artemisApp.quizExercise.addExistingQuestions"></span>
-                    </button>
-                </div>
-            </div>
-            <!--Integrate the template for existingQuestions if the button is clicked-->
-            <ng-container *ngIf="showExistingQuestions; then existingQuestionsTemplate"></ng-container>
-            <ng-container *ngIf="showExistingQuestions && showExistingQuestionsFromCourse; then existingQuestionsFromCourseTemplate"></ng-container>
-            <ng-container *ngIf="showExistingQuestions && showExistingQuestionsFromExam; then existingQuestionsFromExamTemplate"></ng-container>
-
-            <div class="question-options row d-flex justify-content-start" *ngIf="showExistingQuestions && showExistingQuestionsFromFile">
-                <div class="me-2 col flex-grow-0" style="white-space: nowrap">
-                    <span jhiTranslate="artemisApp.quizExercise.importJSON" class="font-weight-bold colon-suffix no-flex-shrink"> </span>
-                </div>
-                <div class="input-group flex-grow-1 col">
-                    <input id="importFileInput" type="file" accept=".json" class="form-control" (change)="setImportFile($event)" placeholder="Upload file..." />
-                    <button class="btn btn-outline-primary" (click)="importQuiz()" jhiTranslate="entity.action.import" [ngClass]="{ disabled: !importFile }"></button>
-                </div>
-            </div>
+            <jhi-quiz-question-list-edit
+                #quizQuestionsEdit
+                [courseId]="courseId!"
+                [quizQuestions]="quizExercise.quizQuestions!"
```

Comment:
How can you be certain that `courseId` and `quizExercise.quizQuestions` are never `null` or `undefined`? I think it would be better to be defensive and check for `null` or `undefined` in the component that accesses them. The default value in `@Input() quizQuestions: QuizQuestion[] = [];` in the `quiz-question-list-edit.component` will be overwritten if the parent component sets `null` or `undefined` as the input.

Reply:
Thanks for noticing! Added a wrapper with ngIf of jhi-quiz-question-list-edit component. Also added a check if quizQuestions is undefined.